### ISSUE TITLE
[Bug]: Cannot operate on any files with HTML entities in their name

### ISSUE THREAD

**Issue by iw0nderhow** [nextcloud/server#43331]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I can't view, move, copy, download, share, delete or open the sidebar for any file whose name contains an HTML entity.

Here's a video including the network pane: https://github.com/nextcloud/server/assets/6024426/d6fa3909-525f-4858-a045-bb8073281ac5

The state in which the viewer opens, but fails to load the file, is already filed in nextcloud/viewer#2095.

### Steps to reproduce

1. Upload a file with a name containing an HTML entity (e.g., `&amp;`) to Nextcloud.
  Alternatively, rename a file to contain an HTML entity.
2. Refresh the page.
3. No operations succeed, at all.

### Expected behavior

I can operate on the file without any issues whatsoever.

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

28

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 22 to 23)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### List of activated Apps

```shell
Enabled:
  - activity: 2.20.0
  - admin_audit: 1.18.0
  - bookmarks: 13.1.3
  - bruteforcesettings: 2.8.0
  - calendar: 4.6.4
  - camerarawpreviews: 0.8.4
  - circles: 28.0.0-dev
  - cloud_federation_api: 1.11.0
  - comments: 1.18.0
  - contacts: 5.5.1
  - contactsinteraction: 1.9.0
  - dashboard: 7.8.0
  - data_request: 1.15.0
  - dav: 1.29.1
  - event_update_notification: 2.3.0
  - external: 5.3.1
  - federatedfilesharing: 1.18.0
  - federation: 1.18.0
  - files: 2.0.0
  - files_automatedtagging: 1.18.0
  - files_external: 1.20.0
  - files_pdfviewer: 2.9.0
  - files_reminders: 1.1.0
  - files_sharing: 1.20.0
  - files_texteditor: 2.15.1
  - files_trashbin: 1.18.0
  - files_versions: 1.21.0
  - firstrunwizard: 2.17.0
  - groupfolders: 16.0.3
  - logreader: 2.13.0
  - lookup_server_connector: 1.16.0
  - maps: 1.3.1
  - news: 24.0.0
  - nextcloud_announcements: 1.17.0
  - notes: 4.9.2
  - notifications: 2.16.0
  - notify_push: 0.6.9
  - oauth2: 1.16.3
  - password_policy: 1.18.0
  - phonetrack: 0.7.7
  - photos: 2.4.0
  - previewgenerator: 5.4.0
  - privacy: 1.12.0
  - provisioning_api: 1.18.0
  - qownnotesapi: 23.12.0
  - recommendations: 2.0.0
  - related_resources: 1.3.0
  - richdocuments: 8.3.1
  - serverinfo: 1.18.0
  - settings: 1.10.1
  - sharebymail: 1.18.0
  - support: 1.11.0
  - survey_client: 1.16.0
  - systemtags: 1.18.0
  - tasks: 0.15.0
  - telephoneprovider: 1.0.3
  - text: 3.9.1
  - theming: 2.3.0
  - twofactor_admin: 4.4.0
  - twofactor_backupcodes: 1.17.0
  - twofactor_totp: 10.0.0-beta.2
  - twofactor_webauthn: 1.3.2
  - updatenotification: 1.18.0
  - user_status: 1.8.1
  - viewer: 2.2.0
  - weather_status: 1.8.0
  - workflowengine: 2.10.0
Disabled:
  - announcementcenter: 6.7.0 (installed 6.7.0)
  - apporder: 0.15.0 (installed 0.15.0)
  - audioplayer: 3.4.1 (installed 3.4.1)
  - checksum: 1.2.3 (installed 1.2.3)
  - customproperties: 2.0.4 (installed 2.0.4)
  - drawio: 3.0.2 (installed 3.0.2)
  - dropit: 0.4.0 (installed 0.4.0)
  - encryption: 2.16.0
  - extract: 1.3.6 (installed 1.3.6)
  - files_downloadactivity: 1.16.0 (installed 1.16.0)
  - files_linkeditor: 1.1.16 (installed 1.1.16)
  - files_markdown: 2.4.1 (installed 2.4.1)
  - files_readmemd: 2.0.1 (installed 2.0.1)
  - files_rightclick: 0.15.1 (installed 1.6.0)
  - integration_mastodon: 2.0.3 (installed 2.0.3)
  - keeporsweep: 0.3.0 (installed 0.3.0)
  - metadata: 0.19.0 (installed 0.19.0)
  - polls: 6.0.1 (installed 6.0.1)
  - ransomware_protection: 1.14.0 (installed 1.14.0)
  - side_menu: 3.11.5 (installed 3.11.5)
  - socialsharing_email: 3.0.1 (installed 3.0.1)
  - spreed: 18.0.3 (installed 18.0.3)
  - suspicious_login: 6.0.0
  - tables: 0.6.6 (installed 0.6.6)
  - twofactor_nextcloud_notification: 3.8.0 (installed 3.8.0)
  - user_ldap: 1.19.0
  - user_saml: 6.1.1 (installed 6.1.1)
  - workflow_script: 1.13.0 (installed 1.13.0)
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Additional info

Tested on Firefox 122.0 and Safari 17.2.1

**Comment by susnux** [comment]
This is a dependency issue of the webdav client we use on the frontend:
https://github.com/perry-mitchell/webdav-client/issues/363

**Comment by perry-mitchell** [comment]
Released `webdav`@`5.3.2` just now, with the above PR merged.
